### ISSUE TITLE
What meaning all the indicators after training?

### ISSUE THREAD

**Issue by JCamiloB21** [matterport/Mask_RCNN#1112]
Hi,
After training the results are 
`Epoch 30/30
25/25 [==============================] - 1797s 72s/step - loss: 1.0709 - rpn_class_loss: 0.0133 - rpn_bbox_loss: 0.3766 - mrcnn_class_loss: 0.2582 - mrcnn_bbox_loss: 0.1942 - mrcnn_mask_loss: 0.2286 - val_loss: 1.0804 - val_rpn_class_loss: 0.0092 - val_rpn_bbox_loss: 0.2886 - val_mrcnn_class_loss: 0.3349 - val_mrcnn_bbox_loss: 0.2035 - val_mrcnn_mask_loss: 0.2441`
but I don't know what is the meaning of each indicator, for example what is the difference between rpn_class_loss and val_rpn_class_loss or loss and val_loss, etc.

thanks

**Comment by stygian2a** [comment]
So, you have 5 "small" losses:
- rpn_class_loss    : How well the Region Proposal Network separates background with objetcs
- rpn_bbox_loss    : How well the RPN localize objects
- mrcnn_bbox_loss : How well the Mask RCNN localize objects
- mrcnn_class_loss : How well the Mask RCNN recognize each class of object
- mrcnn_mask_loss : How well the Mask RCNN segment objects

That makes a bigger loss:
- loss : A combination (surely an addition) of all the smaller losses.

All of  those losses are calculated on the training dataset.

The losses for the validation dataset are those starting with 'val'

Hope this helps.

**Comment by JCamiloB21** [comment]
Thanks!!!!!

**Comment by cpoptic** [comment]
What would cause the val_rpn_bbox_loss to remain high even after training for ~100 epochs?  The other 4 losses are much lower by comparison.   Clearly something with the Region Proposal Network is not localizing the objects well.  Could adjusting the config hyperparameters RPN_ANCHOR_SCALES improve this?

**Comment by Amrimn** [comment]
@cpoptic I have the same issue with the rpn_bbox_loss liss that increases. did you find a solution for that?
Thanks

**Comment by cpoptic** [comment]
Unfortunately no, and this was a main reason why I switched frameworks.

Alternatively it may have to do with the anchor sizes and anchor ratios not
capturing the underlying objects.

The rpn_bbox_loss was the highest component of overall loss for the vision
project was working on.  Followed by the rpn_mask_loss.
The other 3 loss comments however decreased nicely after a hundred or so
epochs of training.

So not sure if there's an issue with the Matterport RPN implementation or
if I'm making an error on my end

**Comment by vijaygill** [comment]
@cpoptic Which framework did you go for then? I am evaluating this repository and here are my graphs. The graphs are truncated because I had to kill and resume the training and my script cleans the tensorboard log data (I have to fix that, but later).

I am running this training on a dataset of 192681 images which, by using augmentation, gets enlarged to 693821 in one iteration and my current session is supposed to run for 2 iterations.

![Untitled](https://user-images.githubusercontent.com/8999486/71207412-ba0fdb80-229e-11ea-8805-3b2d6b6daa2a.png)

**Comment by Amrimn** [comment]
does anyone find solotion for deacreasing  rpn_bbox_loss and mrcnn_bbox_loss?

thanks

**Comment by Kovaxs** [comment]
> @cpoptic Which framework did you go for then? I am evaluating this repository and here are my graphs. The graphs are truncated because I had to kill and resume the training and my script cleans the tensorboard log data (I have to fix that, but later).
> 
> I am running this training on a dataset of 192681 images which, by using augmentation, gets enlarged to 693821 in one iteration and my current session is supposed to run for 2 iterations.
> 
> ![Untitled](https://user-images.githubusercontent.com/8999486/71207412-ba0fdb80-229e-11ea-8805-3b2d6b6daa2a.png)

Excuse me but, how do you have all this graphics losses? I'm only able to have 2: loss and val_loss.

**Comment by Juuustin** [comment]
> So, you have 5 "small" losses:
> 
> * rpn_class_loss    : How well the Region Proposal Network separates background with objetcs
> * rpn_bbox_loss    : How well the RPN localize objects
> * mrcnn_bbox_loss : How well the Mask RCNN localize objects
> * mrcnn_class_loss : How well the Mask RCNN recognize each class of object
> * mrcnn_mask_loss : How well the Mask RCNN segment objects
> 
> That makes a bigger loss:
> 
> * loss : A combination (surely an addition) of all the smaller losses.
> 
> All of those losses are calculated on the training dataset.
> 
> The losses for the validation dataset are those starting with 'val'
> 
> Hope this helps.

Very helpful explanation. I find that the training loss is the sum of those five loss, while the validation loss is not, is that correct?

**Comment by vijaygill** [comment]
> > @cpoptic Which framework did you go for then? I am evaluating this repository and here are my graphs. The graphs are truncated because I had to kill and resume the training and my script cleans the tensorboard log data (I have to fix that, but later).
> > I am running this training on a dataset of 192681 images which, by using augmentation, gets enlarged to 693821 in one iteration and my current session is supposed to run for 2 iterations.
> > ![Untitled](https://user-images.githubusercontent.com/8999486/71207412-ba0fdb80-229e-11ea-8805-3b2d6b6daa2a.png)
> 
> Excuse me but, how do you have all this graphics losses? I'm only able to have 2: loss and val_loss.

Using the regex filter on that page.

**Comment by ApoorvaOjha** [comment]
> @cpoptic Which framework did you go for then? I am evaluating this repository and here are my graphs. The graphs are truncated because I had to kill and resume the training and my script cleans the tensorboard log data (I have to fix that, but later).
> 
> I am running this training on a dataset of 192681 images which, by using augmentation, gets enlarged to 693821 in one iteration and my current session is supposed to run for 2 iterations.
> 
> ![Untitled](https://user-images.githubusercontent.com/8999486/71207412-ba0fdb80-229e-11ea-8805-3b2d6b6daa2a.png)

Hey can you please provide the code how to visualize these losses in tensorboard?

**Comment by agbozo1** [comment]
> > @cpoptic Which framework did you go for then? I am evaluating this repository and here are my graphs. The graphs are truncated because I had to kill and resume the training and my script cleans the tensorboard log data (I have to fix that, but later).
> > I am running this training on a dataset of 192681 images which, by using augmentation, gets enlarged to 693821 in one iteration and my current session is supposed to run for 2 iterations.
> > ![Untitled](https://user-images.githubusercontent.com/8999486/71207412-ba0fdb80-229e-11ea-8805-3b2d6b6daa2a.png)
> 
> Hey can you please provide the code how to visualize these losses in tensorboard?

Here's all you need to do.
Open cmd, activate the python environment (with tensorboard installed)
Go to your project folder and make sure you have your logs folder inside
Then place this code within cmd - 
`tensorboard --logdir=projectFolder/logs/`
Hit Enter - wait for tensorboard to setup and open the localhost address it generates in cmd (copy and paste in your browser).
